core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -486,12 +487,14 @@ private[kafka] class Processor(val id: Int,
   private def processCompletedReceives() {
     selector.completedReceives.asScala.foreach { receive =>
       try {
-        val channel = selector.channel(receive.source)
+        val openChannel = selector.channel(receive.source)
+        val channel = if (openChannel != null) openChannel else selector.closedChannel(receive.source)
         val session = RequestChannel.Session(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, channel.principal.getName),
           channel.socketAddress)
         val req = RequestChannel.Request(processor = id, connectionId = receive.source, session = session, buffer = receive.payload, startTimeMs = time.milliseconds, securityProtocol = protocol)
         requestChannel.sendRequest(req)
-        selector.mute(receive.source)
+        if (openChannel != null)
+            selector.mute(openChannel.id)
```

Comment:
The check was necessary to avoid muting closed channels, but the other change was unnecessary, I have reverted. Thanks.